### PR TITLE
⚠ Remove DelegatedClient, move Options in client.New

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@redhat.com>

### CODE REVIEW

**Comment by vincepri** [comment]
/retest 

flake

File: pkg/manager/manager.go [hunk 1]
```diff
@@ -264,7 +264,7 @@ type Options struct {
 	// NewClient is the func that creates the client to be used by the manager.
 	// If not set this will create the default DelegatingClient that will
 	// use the cache for reads and the client for writes.
-	NewClient cluster.NewClientFunc
+	NewClient client.NewClientFunc
```
Comment (FillZpp) [inline pkg/manager/manager.go:267]: The comment that says `DelegatingClient` should be updated.

File: pkg/cluster/cluster.go [hunk 2]
```diff
@@ -108,7 +108,7 @@ type Options struct {
 	// If not set this will create the default DelegatingClient that will
 	// use the cache for reads and the client for writes.
 	// NOTE: The default client will not cache Unstructured.
-	NewClient NewClientFunc
+	NewClient client.NewClientFunc
```
Comment (FillZpp) [inline pkg/cluster/cluster.go:111]: The comment that says `DelegatingClient` should be updated.

**Review by FillZpp — COMMENTED** [review]
Nice change, I have seen lots of ppl confused by the hidden DelegatedClient.

File: pkg/cluster/cluster.go (hunk 2 above)
Comment (vincepri) [reply pkg/cluster/cluster.go:111]: fixed, thanks!

File: pkg/manager/manager.go (hunk 1 above)
Comment (vincepri) [reply pkg/manager/manager.go:267]: fixed, thanks!

**Pull request merged**